File: Source/Library.Core/Entities/Book.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace Library.Core.Entities
{
    /// <summary>
    /// Represents a book from the library.
    /// </summary>
    public class Book : BaseEntity<Guid>
    {
        public Book()
        {
            Categories = new List<Category>();
            Authors = new List<Author>();
            BookStockByEditions = new List<BookStockByEdition>();
        }

        /// <summary>
        /// Holds the book's title.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Holds the book's publication date.
        /// </summary>
        public DateTime PublicationDate { get; set; }

        /// <summary>
        /// Holds the book's categories information.
        /// </summary>
        public virtual ICollection<Category> Categories { get; set; }

        /// <summary>
        /// Holds the book's authors information.
        /// </summary>
        public virtual ICollection<Author> Authors { get; set; }

        /// <summary>
        /// Holds the library stock for the books by editions's authors information.
        /// </summary>
        public virtual ICollection<BookStockByEdition> BookStockByEditions { get; set; }
    }
}

File: Source/Library.Vuejs/src/models/Book.ts
import { Guid } from "guid-typescript";
import { Author } from "./Author";
import { Category } from "./Category";
import { BookStockByEditions } from "./BookStockByEditions";

export type Book = {
  id: Guid;
  title: string;
  publicationDate: Date;
  categories: Category[];
  authors: Author[];
  bookStockByEditions: BookStockByEditions[];
};

export type BookForm = {
  id: Guid;
  stockId: string;
  title: string;
  publicationDate: Date | null;
  bookEdition: number;
  availableBooks: number;
  borrowedBooks: number;
  categories: Guid[];
  authors: Guid[];
};
